### PR TITLE
Feature/tutor assessment tracking

### PR DESCRIPTION
**Pull request by JonasPetry**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I added multiple integration tests (Spring) related to the features
- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I added multiple integration tests (Jest) related to the features
- [x] Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

Currently, changes of automatic feedback in text-exercises is not tracked. This PR introduces the connection to an external tracking component placed in Athene. There the data is collected and will be analysed at a later point. This way there are no modifications to the Artemis database scheme and it has no impact on the performance of the Artemis server.

#### Server
- Add profile "athene"
- Create a jwt token and add an extra header to the assessment route when profile "athene" is set

#### Client
- Store the jwt token in the participation
- Send a tracking request (Fire & Forget) with the result when the new header is set and the feedback is loaded, saved or submitted
- Route the tracking request directly to athene using nginx **(NOTE: This is done on the server, you cannot see it in the code!)**

### Steps for Testing on Testserver 1 or 3

Start Artemis locally with "ARTEMIS (SERVER & TEXT CLUSTERING)
1. Log in to Artemis
2. Navigate to Course Administration
3. Assess a submission for a text exercise
4. See that tracking request is send in your browser dev tools (There will be an error because nginx is not running locally)
5. See that you don't have a console error

### Nginx Config:
```
location /athene-tracking {
        proxy_pass https://athene01.ase.in.tum.de/tracking;
        proxy_set_header Host $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
```

### CODE REVIEW

File: src/main/webapp/app/exercises/text/assess/text-assessments.service.ts [hunk 1]
```diff
@@ -167,4 +175,32 @@ export class TextAssessmentsService {
...
+        });
+
+        const sendObject = {
+            textBlocks: submissionForSending?.blocks,
+            participation: submissionForSending?.participation,
+        };
+
+        if (submission!.participation?.trackingToken) {
+            this.http
+                .post(`${SERVER_API_URL}/athene-tracking/text-exercise-assessment}`, sendObject, {
```
Comment (krusche) [inline src/main/webapp/app/exercises/text/assess/text-assessments.service.ts:200]: This endpoint does not exist on Artemis. Would it make sense to send this to Athene directly?

File: src/main/webapp/app/exercises/text/assess/text-assessments.service.ts [hunk 2]
```diff
@@ -167,4 +175,32 @@ export class TextAssessmentsService {
...
+            result.participation = null;
+            result.submission = null;
+        });
+
+        const sendObject = {
+            textBlocks: submissionForSending?.blocks,
+            participation: submissionForSending?.participation,
+        };
+
+        if (submission!.participation?.trackingToken) {
```
Comment (krusche) [inline src/main/webapp/app/exercises/text/assess/text-assessments.service.ts:198]: the if statement comes too late. I would check that first in this method

Comment (JonasPetry) [reply src/main/webapp/app/exercises/text/assess/text-assessments.service.ts:198]: Done

File: src/main/webapp/app/exercises/text/assess/text-assessments.service.ts (hunk 1 above)
Comment (JonasPetry) [reply src/main/webapp/app/exercises/text/assess/text-assessments.service.ts:200]: The idea is to route it via nginx directly to athene. I've added an inline comment to make this clear for every developer.

**Comment by JonasPetry** [comment]
@krusche I've adressed all the issues you mentioned.

- Added description about how & where the request is send
- Implemented client & server tests
- Improved the JavaDoc / JSDoc
- Moved the key into the TextSubmission

**Comment by krusche** [comment]
I will take a look as soon as possible. In the mean time, can we get reviews and tests by other developers as we do it usually?

**Pull request merged**